src/Microsoft.DotNet.XHarness.Tests.Runners/NUnit/NUnitTestListener.cs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Xml;
+using Microsoft.DotNet.XHarness.Tests.Runners.Core;
+using NUnit;
+using NUnit.Engine;
+using NUnit.Framework.Interfaces;
+
+#nullable  enable
+namespace Microsoft.DotNet.XHarness.Tests.Runners.NUnit
+{
+    internal class NUnitTestListener : ITestEventListener
+    {
+        readonly LogWriter _logger;
+        readonly INUnitTestRunner _runner;
+
+        public NUnitTestListener(INUnitTestRunner runner, LogWriter logger)
+        {
+            _runner = runner ?? throw new ArgumentNullException(nameof (runner));
+            _logger = logger ?? throw new ArgumentNullException(nameof (logger));
+        }
+
+        void TestStarted(XmlNode testEvent)
+        {
+            if (testEvent == null)
+                return;
+
+			if (!string.IsNullOrEmpty (_runner.TestsRootDirectory))
+				Environment.CurrentDirectory = _runner.TestsRootDirectory;
+
+            _logger.OnInfo (testEvent.Attributes["fullname"].Value);
+
+        }
+
+        void TestFinished(XmlNode testEvent)
+        {
+            // we need to get the status from the string. That value is stored in
+            // <test-case result=''> value can be: Passed, Failed, Inconclusive or Skipped. To make things
+            // 'simpler' we also need to check the label, which might have the following values:  Error, Cancelled or Invalid
+            var result = testEvent.GetAttribute("label") ?? testEvent.GetAttribute("result");
+            TestStatus status = result switch
+            {
+                "Passed" => TestStatus.Passed,
```

Comment:
Just asking, but should this somehow come from the jargon switch that I've seen in other file?